src/prefect/workers/base.py

```diff
@@ -734,30 +739,39 @@ async def sync_with_backend(self):
         queues. Sends a worker heartbeat to the API.
         """
         await self._update_local_work_pool_info()
-        # Only do this logic if we've enabled the experiment, are connected to cloud and we don't have an ID.
-        if (
-            get_current_settings().experiments.worker_logging_to_api_enabled
-            and (
-                self._client.server_type == ServerType.CLOUD
-                or get_current_settings().testing.test_mode
+        try:
+            remote_id = await self._send_worker_heartbeat(
+                get_worker_id=(self._should_get_worker_id())
             )
-            and self.backend_id is None
-        ):
-            get_worker_id = True
-        else:
-            get_worker_id = False
-
-        remote_id = await self._send_worker_heartbeat(get_worker_id=get_worker_id)
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 422 and self._should_get_worker_id():
+                self._logger.warning(
+                    "Failed to retrieve worker ID from the Prefect API server."
+                )
+                await self._send_worker_heartbeat(get_worker_id=False)
+                remote_id = None
+            else:
+                raise e
 
-        if get_worker_id and remote_id is None:
+        if self._should_get_worker_id() and remote_id is None:
             self._logger.warning(
                 "Failed to retrieve worker ID from the Prefect API server."
             )
-        else:
-            self.backend_id = remote_id
+        elif self.worker_id is None and remote_id is not None:
+            self.worker_id = remote_id
+            self._logger = get_worker_logger(self)
+            self._logger.info(f"id {self.worker_id}")
```

Comment:
This should probably be a `DEBUG` log.

Reply:
i'll nix that, was from the demo. Sorry about that